.ci/docker/requirements-dev.txt

```diff
@@ -1,6 +1,16 @@
+datasets >= 3.6.0
+einops
 expecttest==0.1.6
+pillow
 pytest==7.3.2
 pytest-cov
 pre-commit
+pyrefly==0.45.1
 tomli-w >= 1.1.0
+torch
```

Comment:
We can not install pytorch using pip install, as torchtitan depends on pytorch-nightly: https://github.com/pytorch/torchtitan/blob/refs/heads/main/README.md#nightly-builds